code-studio/src/js/callouts.js

```diff
@@ -112,16 +124,49 @@ module.exports = function createCallouts(callouts) {
     }
 
     if (callout.on) {
-      $(window).on(callout.on, function () {
-        if (!callout.seen && $(selector).length > 0) {
+      $(window).on(callout.on, function (e, action) {
+        if (!config.codeStudio.selector) {
+          config.codeStudio.selector = selector;
+        }
+        var lastSelector = config.codeStudio.selector;
+        $(window).trigger('prepareforcallout', [config.codeStudio]);
+        if (lastSelector !== config.codeStudio.selector && $(lastSelector).length > 0) {
+          $(lastSelector).qtip(config).qtip('destroy');
+        }
+        if ($(config.codeStudio.selector).length > 0) {
+          if (action === 'hashchange' || action === 'hashinit' || !callout.seen) {
+            $(config.codeStudio.selector).qtip(config).qtip('show');
+          }
           callout.seen = true;
-          $(selector).qtip(config).qtip('show');
         }
       });
-    } else {
+    } else if (!callout.seen) {
       $(selector).qtip(config).qtip('show');
     }
   });
+
+  // Insert a hashchange handler to detect triggercallout= hashes and fire
+  // appropriate events to open the callout
+  function detectTriggerCalloutOnHash(event) {
```

Comment:
It seems like one option would be to do something like https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/instructions/MarkdownInstructions.jsx#L59. In that case, after the component has mounted (i.e. there's a DOM representation) we look for any `details` elements and add some jquery handling to them.

We could also search for any descendants with an href containing "#triggercallout=" and for each instance we find, add a click event that triggers the appropriate name. The result would be that we don't have to listen to globally listen to hashchange events for this one feature, and we also dont need to worry about the current state of the hash (since the hash has just become a way to communicate to the instructions component what even it should trigger on click).

This would look something like:
```
$(ReactDOM.findDOMNode(this)).find("a[href*='#triggercallout=]").each((index, item) => {
   const href = item.attr('href');
   // extract trigger callout from href

   item.click(() => { $window.trigger(eventName, [eventType]); });
})
```

Also cc'ing @islemaster to see if he has any thoughts on the best way of accomplishing this, since he's good at coming up with clean ways of doing things :)

Regardless of which approach we take, this feature could also use some sort of tests. This feels like a code path that won't get a ton of coverage otherwise, and would be easy for us to regress and not realize.